Extension/lib/filter/rules/url-filter.js

```diff
@@ -86,29 +86,50 @@ UrlFilter.prototype = {
 	 * @param refHost       Referrer host
 	 * @param requestType   Request content type (UrlFilterRule.contentTypes)
 	 * @param thirdParty    true if request is third-party
+	 * @param genericUrlBlockRule    genericblock rule or null
 	 * @return First matching rule or null if no match found
 	 */
-	isFiltered: function (url, refHost, requestType, thirdParty) {
+	isFiltered: function (url, refHost, requestType, thirdParty, genericUrlBlockRule) {
 		var rule;
+		var genericBlockRuleFired;
+
+		var _isFiltered = function (url, refHost, rules, requestType, thirdParty, genericRulesAllowed) {

```

Comment:
This is pretty bad from the cpu and memory usage point of view.

It would be much better to have a prototype function instead.